test/framework/src/main/java/org/elasticsearch/transport/AbstractSimpleTransportTestCase.java

```diff
@@ -1721,4 +1726,44 @@ public void testRegisterHandlerTwice() {
         serviceA.registerRequestHandler("action1", TestRequest::new, randomFrom(ThreadPool.Names.SAME, ThreadPool.Names.GENERIC),
             (request, message) -> {throw new AssertionError("boom");});
     }
+
+    public void testTimeoutPerConnection() throws IOException {
```

Comment:
We discussed this offline. I'm good with getting this in as is, and we'll follow up on the status of a Mac in CI.